mangum/adapter.py

```diff
@@ -44,15 +43,18 @@ class Mangum:
     * **text_mime_types** - A list of MIME types to include with the defaults that
     should not return a binary response in API Gateway.
     * **dsn** - A connection string required to configure a supported WebSocket backend.
+    * **api_gateway_base_path** - A string specifying the part of the url path after
+    which the server routing begins.
     * **api_gateway_endpoint_url** - A string endpoint url to use for API Gateway when
     sending data to WebSocket connections. Default is to determine this automatically.
     * **api_gateway_region_name** - A string region name to use for API Gateway when
     sending data to WebSocket connections. Default is `AWS_REGION` environment variable.
     """
 
     app: ASGIApp
-    lifespan: str = "auto"
+    lifespan: str
     dsn: Optional[str] = None
+    api_gateway_base_path: str
     api_gateway_endpoint_url: Optional[str] = None
     api_gateway_region_name: Optional[str] = None
```

Comment:
These definitions are leftover from when the adapter was using `dataclass` and should be removed.

Reply:
Are you open to dropping 3.6 and making them `dataclasses` again? Security support was dropped for it a month ago and from what I see 3.6 support was re-added 2 years ago because of Zeit Now which supports ^3.7. I have a branch where I dropped 3.6 and added tests for 3.9 and 3.10 and they all pass.

Reply:
Obviously that's out of scope for this PR.

Reply:
Yep, since 3.6 is now EOL it should probably be looked at, but this is unrelated to `dataclasses` being used or not because they were supported in 3.6 using the backport. I forget when/why it became a normal class again (probably included as part of some larger refactoring), not sure if there is a significant advantage to changing it back or not.